packages/@pollyjs/adapter-puppeteer/src/index.js

```diff
@@ -223,6 +223,19 @@ export default class PuppeteerAdapter extends Adapter {
     if (this[LISTENERS].has(target)) {
       const listeners = this[LISTENERS].get(target);
 
+      // puppeteer remove prependListener after v4.0.0, add it if missing
+      if (!target.prependListener) {
+        target.prependListener = function(event, handler) {
```

Comment:
Instead of mutating `target`, you can do something like this:

`const prependListener = target.prependListener || prependListenerPolyfill;`

then

```js
for (const eventName in listeners) {
  const fn = methodName === 'prependListener' ? prependListener : target[methodName];
  
  fn.apply(target, [eventName, listeners[eventName]]);
}
```

Reply:
got it